src/Text/Pandoc/Readers/LaTeX/Math.hs

```diff
@@ -140,6 +140,15 @@ newtheorem inline = do
                             M.insert name spec tmap }
   return mempty
 
+extractLabelFromBlock :: Block -> Maybe Text
+extractLabelFromBlock (Para inlines) = extractLabel inlines
+  where
+    extractLabel :: [Inline] -> Maybe Text
+    extractLabel (Span (_, _, attrs) _:_) = lookup "label" attrs
+    extractLabel (_:xs) = extractLabel xs
+    extractLabel [] = Nothing
```

Comment:
Won't this return Nothing if there is a span that isn't a label, even if there's a ltr span that *is* a label?
I think a better approach would be a fold, something like
```hs
extractLabel = foldl' go
go :: Maybe Text -> Inline -> Maybe Text
go (Just t) _ = Just t
go Nothing (Span...) = ... lookup as before ...
go Nothing _ = Nothing
````

Reply:
Very true. Updated.